### ISSUE TITLE
Must activities be uniquely named?

### ISSUE THREAD

**Issue by frank-klein-schaarsberg-wb** [TUDelft-CITG/OpenCLSim#153]
Title says it all... Activities can be initialized with a name, but they also always have an ID. It would suffice for only the IDs to be unique, and not the names.

E.g. the following setup: 
- use multiple vessels
- to transport different types of cargo
- traversing a river of multiple segments downstream to a mother port and back upstream to the feeder port

For activities to be uniquely named, traversing a river segment must be named in a following manner:
f"{vessel.name} traverse river segment {segment.name} {direction} carrying {cargo}"

Whereas for the sake of logging it suffices to simply name it f"traverse segment {segment.name}"

So in summary, can we relax the need for uniquely named activities if the ID is already unique?

**Comment by SiggyF** [comment]
I checked the source code and found that processes are registered both by and name: 
https://github.com/TUDelft-CITG/OpenCLSim/blob/92411c260574136bdc6eb7f271516a1cd67fab06/src/openclsim/model/base_activities.py#L157

I think it's a good idea to only register by Id. But I can't directly oversee the consequences.
